dashboard/app/controllers/omniauth_callbacks_controller.rb

```diff
@@ -24,7 +24,11 @@ def all
       silent_takeover(@user)
     elsif User.find_by_email_or_hashed_email(@user.email)
       # Note that @user.email is populated by User.from_omniauth even for students
-      redirect_to "/users/sign_in?providerNotLinked=#{@user.provider}&email=#{@user.email}"
+      if User.find_by_email_or_hashed_email(@user.email).provider == 'clever'
```

Comment:
There might be some Ruby optimization I don't know about, but is it bad to call `User.find_by_email_or_hashed_email` twice in a row here?  Unless the result of that call is cached, we should probably store its first result and use it here.

Reply:
Is there a constant available somewhere we could use in place of the string literal `'clever'`?